python/salpytools/salpylib.py

```diff
@@ -34,42 +34,43 @@
-    face it is intended to be used this way. In essence this gives the 
-    implementer full control on what the recieved call on the model. 
+    When creating DDSControllers a Context is passed into it. When the
+    DDSController recieves commands it delegates these actions to the Context.
+    The context then delegates this to the active State. The salpylib allows
+    the states to be inherited from to override behavior of these states, in
+    face it is intended to be used this way. In essence this gives the
+    implementer full control on what the recieved call on the model.
 
     Attributes:
-        subsystem_tag: A string of the name of the CSC we want. Must exactly 
+        subsystem_tag: A string of the name of the CSC we want. Must exactly
         match the Subsystem Tag defined in XML. Link to current Subsystem Tags
         https://stash.lsstcorp.org/projects/TS/repos/ts_xml/browse/sal_interfaces
         default_state: default state defined within states.py of this library.
         tsleep: A sleeper to prevent race conditions when sending log events.
         states: List of states this context will contain. In general use, these
-        should be subclassed states with overriden behavior. 
+        should be subclassed states with overriden behavior.
     """
-    def __init__(self, subsystem_tag, default_state='OFFLINE', tsleep=0.5, 
+    def __init__(self, subsystem_tag, model, default_state='OFFLINE', tsleep=0.5,
                  states=None):
-        
+
         self.subsystem_tag = subsystem_tag
+        self.model = model
         self.current_state = default_state
         self.tsleep = tsleep
         if states is not None:
             self.states = states
         else:
             # Loading default states
             self.states = dict()
-            self.states["OFFLINE"] = csc_states.BaseState('OFFLINE', self.subsystem_tag)
-            self.states["STANDBY"] = csc_states.BaseState('STANDBY', self.subsystem_tag)
-            self.states["DISABLE"] = csc_states.BaseState('DISABLE', self.subsystem_tag)
-            self.states["ENABLE"] = csc_states.BaseState('ENABLE', self.subsystem_tag)
-            self.states["FAULT"] = csc_states.BaseState('FAULT', self.subsystem_tag)
-            self.states["INITIAL"] = csc_states.BaseState('INITIAL', self.subsystem_tag)
-            self.states["FINAL"] = csc_states.BaseState('FINAL', self.subsystem_tag)
+            self.states["OFFLINE"] = csc_states.OfflineState(self.subsystem_tag)
+            self.states["STANDBY"] = csc_states.StandbyState(self.subsystem_tag)
+            self.states["DISABLED"] = csc_states.DisabledState(self.subsystem_tag)
+            self.states["ENABLED"] = csc_states.EnabledState(self.subsystem_tag)
+            self.states["FAULT"] = csc_states.FaultState(self.subsystem_tag)
+            self.states["INITIAL"] = self.states["STANDBY"]
+            self.states["FINAL"] = self.states["ENABLED"]
```

Comment:
Is this right? INITIAL state is STANDBY? not OFFLINE? And FINAL state is ENABLED?

Reply:
I changed both INITIAL, and FINAL to OFFLINE state. At least this is what I think it is when looking at the state machine diagram on LTS-666?